File: Sc.Models/Dtos/Auth/SignUpDto.cs
using Sc.Models.Dtos.Companies;
using Sc.Models.Dtos.Nomenclatures.Complexes;
using Sc.Models.Dtos.Nomenclatures.Institutions;
using Sc.Models.Dtos.Sso;
using Sc.Models.Enums.Auth;
using Sc.Models.Enums.Suppliers;

namespace Sc.Models.Dtos.Auth
{
    public class SignUpDto
    {
        public string RecaptchaToken { get; set; }

        public SignUpType Type { get; set; }
        public SupplierType SupplierType { get; set; }
        public SupplierExtendedType SupplierExtendedType { get; set; }

        public SsoUserDto User { get; set; }

        // Type = Supplier and SupplierType = Institution
        public InstitutionDto RootInstitution { get; set; }
        public InstitutionDto Institution { get; set; }
        // Type = Supplier and SupplierType = Complex
        public ComplexDto Complex { get; set; }

        // Type = Company
        public CompanyDto Company { get; set; }

        // Info only
        public SsoUserValidateSignUpInfoDto SsoUserValidateSignUpInfo { get; set; }
    }
}

File: WebApplication/src/auth/dtos/sign-up/sign-up.dto.ts
import { CompanyDto } from "src/companies/dtos/company.dto";
import { SsoUserDto } from "./sso-user.dto";
import { SignUpType } from "src/auth/enums/sign-up-type.enum";
import { InstitutionDto } from "src/nomenclatures/dtos/institutions/institution.dto";
import { SsoUserValidateSignUpInfoDto } from "./sso-user-validate-sign-up-info.dto";
import { SupplierType } from "src/suppliers/enums/supplier-type.enum";
import { ComplexDto } from "src/nomenclatures/dtos/complexes/complex.dto";
import { SupplierExtendedType } from "src/auth/enums/supplier-extended-type.enum";

export class SignUpDto {
    recaptchaToken: string;

    type = SignUpType.supplier;
    supplierType = SupplierType.institution;
    supplierExtendedType = SupplierExtendedType.university;

    user: SsoUserDto = new SsoUserDto();

    institution: InstitutionDto = null;
    complex: ComplexDto = null;

    company: CompanyDto = new CompanyDto();

    // Client only
    rootInstitution: InstitutionDto = null;

    // Info only
    ssoUserValidateSignUpInfo: SsoUserValidateSignUpInfoDto = new SsoUserValidateSignUpInfoDto();
}